dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/CompanyUploader.kt

```diff
@@ -126,4 +127,33 @@ class CompanyUploader(
             )
         }
     }
+
+    /**
+     * Updates the ISINs of all companies.
+     * @param leiIsinMap the delta-map with the format "LEI"->"ISIN1,ISIN2,..."
+     */
+    fun updateIsinMapping(leiIsinMap: Map<String, String>) {
+        retryOnCommonApiErrors {
+            for ((lei, isins) in leiIsinMap) {
+                val isinList = isins.split(",").map { it.trim() }
+                val company = companyDataControllerApi.getCompaniesBySearchString(lei)
+                if (company.isNotEmpty()) {
+                    val companyId = company.first().companyId
+
+                    val existingIdentifiers = companyDataControllerApi.getCompanyById(companyId)
+                        .companyInformation.identifiers
+                    val updatedIdentifiers = existingIdentifiers.toMutableMap()
+                    updatedIdentifiers["isin"] = isinList
+
+                    val companyPatch = CompanyInformationPatch(identifiers = updatedIdentifiers)
+                    companyDataControllerApi.patchCompanyById(
+                        companyId,
+                        companyPatch,
+                    )
+                } else {
+                    logger.info("No company found for LEI: $lei")
```

Comment:
Should this rather be an exception? Because in this case we made an erro rsomewhere when importing the companies from the original GLEIF ingestion.